### ISSUE TITLE
FAQ Reader: make tstamp of item accessible within reader template

### ISSUE THREAD

**Issue by reluem** [contao/contao#221]
**Description**  
The current FAQ Reader module outputs a wrong tstamp via $this->tstamp. The reader module always shows the tstamp of the module itself, not of the faq entry it is displaying.

The only way to currently access the tstamp of a FAQ entry is to either manipulate the [$this->info ](https://github.com/contao/faq-bundle/blob/c12bfef2eb354d141c7e1ede97d36a33e2edac0e/src/Resources/contao/modules/ModuleFaqReader.php#L152)output via the default.php language file, or to do an additional request (see [here](https://community.contao.org/de/showthread.php?72849-FAQ-tstamp-im-Reader-ausgeben&p=489534&viewfull=1#post489534)).

While it might be true that the FAQ and News module differ in their structure (the latter having it's own output template like news_full.html5), it would still be great if the tstamp could be added to the module object.

**Example**  
compare output of $this->tstamp in a newsreader (news_full.html5) vs. faqreader module (mod_faqreader.html5)
The tstamp output in the faqreader will be different from the tstamp in the tl_faq table.

**Comment by fritzmg** [comment]
> The current FAQ Reader module outputs a wrong tstamp via $this->tstamp.

It's not exactly wrong, as already explained on the community forum. In any `mod_` template, you have access to the _module's_ data via `$this`.

> compare output of $this->tstamp in a newsreader (news_full.html5) vs. faqreader module (mod_faqreader.html5)

Those are two completely different types of templates. One is a module template and the other one is an "element" template. The FAQ bundle currently does not have the latter.

**Comment by reluem** [comment]
> It's not exactly wrong, as already explained on the community forum. In any mod_ template, you have access to the module's data via $this. [...] Those are two completely different types of templates

Yes, agree, they are different templates. I wanted to illustrate how the expected functionality (IMO) should act.

Wouldn't it be easier anyways (more flexible at least) to decouple the author and tstamp from $this->info and provide them as separate keys in the module's data? 

If I'm the only one questioning this, I am also fine using your proposed solution.

**Comment by leofeyer** [comment]
We should discuss how to handle this case in general. Maybe we can set `$this->Template->model = $objElement` or something similar.

**Comment by leofeyer** [comment]
As discussed in Mumble on January 17th, we want to add [`$this->Template->faq`](https://github.com/contao/contao/blob/ef0dae7c933477932e902af1e2e56ffc8014f4e0/faq-bundle/src/Resources/contao/modules/ModuleFaqList.php#L126) in the `ModuleFaqReader` class as well.

Does anyone know about other templates that miss the original record?

**Comment by leofeyer** [comment]
@contao/developers I'm not 100% sure how to implement this. In the FAQ list and FAQ page modules, `$this->Template->faq` is an array of arrays:

```php
$this->Template->faq = [
    [
        'id' => 1,
        'question' => '…',
        'href' => '…',
    ],
    [
        'id' => 2,
        'question' => '…',
        'href' => '…',
    ]
];
```

Which of the following would you prefer?

```php
// Only the record as object
$this->Template->faq = $objFaq;

// Array of arrays, just like in the other two modules
$this->Template->faq = array($objFaq->row());

// Only the record as array
$this->Template->faq = $objFaq->row();
```

**Comment by fritzmg** [comment]
Imho 1 or 3

**Comment by ausi** [comment]
I would prefer 3 (`$objFaq->row()`)

**Comment by leofeyer** [comment]
Implemented in 1caf237e78691ed12a823efd8a62f66ae0b9fffd.
